internal/statesync/reactor.go

```diff
@@ -345,6 +355,13 @@ func (r *Reactor) Sync(ctx context.Context) (sm.State, error) {
 	return state, nil
 }
 
+func (r *Reactor) publishCommitEvent(commit *types.Commit) error {
+	if r.csState == nil {
+		return nil
+	}
+	return r.csState.PublishCommitEvent(commit)
```

Comment:
Maybe we should publish it to  consensus.State.internalMsgQueue ? This should invoke all the logic that will finally fire the event.

Reply:
using `internalMsgQueue` leads to the code blocking